main.go

```diff
@@ -1851,6 +1878,72 @@ func (git *repoSync) CallAskPassURL(ctx context.Context) error {
 	return nil
 }
 
+func (git *repoSync) RefreshAppToken(ctx context.Context, githubBaseURL, privateKey string, appID, installationID int) error {
+	git.log.V(3).Info("refreshing GitHub app token")
+
+	pkey, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(privateKey))
+	if err != nil {
+		return err
+	}
+
+	now := time.Now()
+
+	claims := jwt.RegisteredClaims{
+		Issuer:    fmt.Sprintf("%d", appID),
+		IssuedAt:  jwt.NewNumericDate(now),
+		ExpiresAt: jwt.NewNumericDate(now.Add(10 * time.Minute)),
+	}
+
+	jwt, err := jwt.NewWithClaims(jwt.SigningMethodRS256, claims).SignedString(pkey)
+	if err != nil {
+		return err
+	}
+
+	url := fmt.Sprintf("https://%s/app/installations/%d/access_tokens", githubBaseURL, installationID)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Authorization", "Bearer "+jwt)
+	req.Header.Set("Accept", "application/vnd.github+json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+	if resp.StatusCode != 201 {
+		errMessage, err := io.ReadAll(resp.Body)
```

Comment:
Shouldn't we bound the read here to 1K or 4K or 32K or something?

Reply:
That would make sense, should the same apply to the `CallAskPassURL` usage of `io.ReadAll`?

Reply:
Missed this before, yeah, it should :)

Reply:
You know what, this is fine.  I'm just being paranoid.  Doing a bounded copy is nice, but I won't require it here.